### ISSUE TITLE
VFS for views

### ISSUE THREAD

**Issue by shimaore** [pugjs/pug#682]
With the introduction of Express 3, the responsibility for view lookups and loading partials (such as `include` in Jade) has migrated from Express to the templating engine. However currently there's no way for a framework built over Express 3 to tell Jade (or other templating engines) how to access templates that might be store elsewhere than on the filesystem. In Jade, `renderFile()` and `parseInclude()` rely directlly on `fs.readFileSync`. (The `consolidate` module makes the same assumption.)

It would be good to extend the template engine API to allow for the provision of a third-party accessor; since Jade is the first template engine to provide `__express` (and is developped by @visionmedia , like Express), I was thinking it might be the best place to specify/attempt to do this.

Basically what we're looking for (in our case for zappajs/zappajs) is to be able to provide to a template engine a function (either sync or async, it doesn't really matter in our case) that provides:

```
readSync(path) { return content }
/* or */
read(path,callback) { callback(err,content) }
```

and replace the engines (or `consolidate`'s) `fs.readFile[Sync]` calls.

This way we can provide access to in-memory template content; or other frameworks could think of other ways to do this (e.g. load the template/included text from a remote document).
